Source/SpatialGDK/Private/SpatialInterop.cpp

```diff
@@ -613,30 +613,38 @@ void USpatialInterop::ResolvePendingIncomingObjectUpdates(UObject* Object, const
 		USpatialActorChannel* DependentChannel = ChannelProperties.Key;
 		FPendingIncomingProperties& Properties = ChannelProperties.Value;
 
-		// Trigger pending updates.
-		DependentChannel->PreReceiveSpatialUpdate(Object);
-
+		// Replicated Properties
 		TSet<UProperty*> RepNotifies;
-		for (const FRepHandleData* RepData : Properties.Key)
+		for (auto& ObjsToReplicate : Properties.Key)  // TPair<UObject*, TArray<const FRepHandleData*>>
 		{
-			ApplyIncomingReplicatedPropertyUpdate(*RepData, DependentChannel->Actor, &Object, RepNotifies);
-			UE_LOG(LogSpatialGDKInterop, Log, TEXT("%s: Received queued object replicated property update. actor %s (%lld), property %s"),
-				*SpatialOSInstance->GetWorkerId(),
-				*DependentChannel->Actor->GetName(),
-				DependentChannel->GetEntityId().ToSpatialEntityId(),
-				*RepData->Property->GetName());
+			// Iterate through our list of target objects, either an actor or component
+			// For each, we resolve all replicated properties under that object
+			UObject* TargetObject = ObjsToReplicate.Key;
+			DependentChannel->PreReceiveSpatialUpdate(TargetObject);
+
+			for (const FRepHandleData* RepData : ObjsToReplicate.Value) {  // Iterating through TArray<const FRepHandleData*>
+				ApplyIncomingReplicatedPropertyUpdate(*RepData, TargetObject, &IncomingObject, RepNotifies);
+				UE_LOG(LogSpatialGDKInterop, Log, TEXT("%s: Received queued object replicated property update. Channel actor (root actor) %s (%lld), target object %s, property %s"),
+					*SpatialOSInstance->GetWorkerId(),
+					*DependentChannel->Actor->GetName(),
+					DependentChannel->GetEntityId().ToSpatialEntityId(),
+					*TargetObject->GetName(),
+					*RepData->Property->GetName());
+			}
+
+			DependentChannel->PostReceiveSpatialUpdate(TargetObject, RepNotifies.Array());
 		}
+
+		// Handover
 		for (const FHandoverHandleData* HandoverData : Properties.Value)
 		{
-			ApplyIncomingHandoverPropertyUpdate(*HandoverData, DependentChannel->Actor, &Object);
-			UE_LOG(LogSpatialGDKInterop, Log, TEXT("%s: Received queued object handover property update. actor %s (%lld), property %s"),
+			ApplyIncomingHandoverPropertyUpdate(*HandoverData, DependentChannel->Actor, &IncomingObject);  // HandoverHandleData will find the correct TargetObject for our replicated property, so we can just pass in the root actor
```

Comment:
Channel actor, or owning actor instead of root actor.

Reply:
done